lib/connextclient/ConnextClient.ts

```diff
@@ -1045,13 +1084,33 @@ class ConnextClient extends SwapClient {
       units = argUnits.toString();
     }
 
-    const res = await this.sendRequest('/onchain-transfer', 'POST', {
-      assetId: this.getTokenAddress(currency),
-      amount: units,
-      recipient: destination,
+    if (!this.seed) {
+      throw new Error('cannot broadcast transaction without seed');
+    }
+
+    // This is a bridge between the imperative and functional code. First step is to get rid of all of the external state.
+    const broadcastTransaction$ = onChainSendERC20(
+      this.host,
+      this.port,
+      CHAIN_IDENTIFIERS[this.network],
+      this.seed,
+      this.getTokenAddress(currency),
+      destination
+    );
```

Comment:
Aren't you missing the `units: string` argument here?

Reply:
Yes, indeed :)